src/sql/relation.rs

```diff
@@ -251,6 +251,7 @@ impl<'a, T: QueryToRelationTranslator + Copy + Clone> VisitedQueryRelations<'a,
     fn try_from_table_with_joins(
```

Comment:
This function is becoming too complex, can you try to split it in independent sub-functions with explicit names ?